### DISCUSSION TITLE
In trap mode, the trap policy remains in the kernel.

### DISCUSSION THREAD

**Discussion by chenjinxing** [strongswan/strongswan#959] [General]
**Describe the bug**
When the "start_action = trap" option is used and the child-sa is created, Strongswan creates two groups of policies with different priorities. However, after the SA is disabled and an empty configuration is reloaded, the trap policy remains in the system.

**To Reproduce**
1. Set "start_action = trap" in the swanctl.conf file.

2. Run the command "swanctl -q -f" load swanctl.conf，Run the command "swanctl -i -c" create child-sa. Strongswan creates two 
groups of policies with different priorities.

3. Use "swanctl -t -c" close child-sa，use "swanctl -q -c" reload empty conf，Run the command "ip xfrm policy" show all policy, the trap policy remains in the system.

**Expected behavior**
strongswan needs to delete all policies added by itself.

**Additional context**
kernel_netlink_ipsec.c: del_policy
the mapping to SA has been delete by "used_by->remove_at", but when use_count > 0, strongswan does not send netlink messages to the kernel.

**Comment by tobiasbrunner** [comment]
I can't reproduce this. Please provide logs (see #196).

**Edit:** This might actually be the same issue as discussed here: https://github.com/strongswan/strongswan/discussions/989#discussioncomment-2524231
